Source/WebCore/rendering/RenderWidget.cpp

```diff
@@ -228,8 +228,10 @@ void RenderWidget::styleDidChange(StyleDifference diff, const RenderStyle* oldSt
         else
             m_widget->show();
 
-        if (CheckedPtr cache = document().existingAXObjectCache())
+        if (CheckedPtr cache = document().existingAXObjectCache()) {
             cache->onWidgetVisibilityChanged(*this);
+            cache->childrenChanged(this);
+        }
     }
 }
 
```

Comment:
In addition, we should not call cache->childrenChanged and onWidgetVisibilityChanged. We should have only one call to an AXObjectCache event listener, and sort out in there what needs to be updated. Maybe we need to pass more context info to this onWidgetChanged listener to determine what to do.